### PR TITLE
Prefill membership form

### PR DESCRIPTION
**Pull request by KaiNissen**
This is mainly for prefilling the membership application form embedded into the donation confirmation page with the data the donor entered for their donation. I also improved the route tests to check more of the data available to templates.

This PR tackles the last bits of [phab:T135110](https://phabricator.wikimedia.org/T135110).

### CODE REVIEW

File: src/Presentation/Presenters/DonationConfirmationHtmlPresenter.php [hunk 1]
```diff
@@ -87,4 +92,37 @@ private function getBankDataArguments( PaymentMethod $paymentMethod ): array {
...
\ No newline at end of file
+	private function getInitialMembershipFormValues( Donation $donation ) {
+		return array_merge(
+			$this->getMembershipFormPersonValues( $donation->getDonor() ),
+			$this->getMembershipFormBankDataValues( $donation->getPaymentMethod() )
+		);
+	}
+
+	private function getMembershipFormPersonValues( Donor $donor ) {
+		return $donor !== null ? [
```
Comment (JeroenDeDauw) [inline src/Presentation/Presenters/DonationConfirmationHtmlPresenter.php]: Since the type hint is not nullable, the null case cannot occur here

**Comment by JeroenDeDauw** [comment]
+1 apart from the bug

**Comment by JeroenDeDauw** [comment]
Is this https://phabricator.wikimedia.org/T135110 ?

File: src/Presentation/Presenters/DonationConfirmationHtmlPresenter.php (hunk 1 above)
Comment (KaiNissen) [inline src/Presentation/Presenters/DonationConfirmationHtmlPresenter.php]: You're right, that improves readability.

File: src/Presentation/Presenters/DonationConfirmationHtmlPresenter.php [hunk 2]
```diff
@@ -87,4 +91,49 @@ private function getBankDataArguments( PaymentMethod $paymentMethod ): array {
...
+			$this->getMembershipFormPersonValues( $donation->getDonor() ),
+			$this->getMembershipFormBankDataValues( $donation->getPaymentMethod() )
+		);
+	}
+
+	/**
+	 * @param Donor|null $donor
+	 * @return array
+	 */
+	private function getMembershipFormPersonValues( $donor ) {
```
Comment (JeroenDeDauw) [inline src/Presentation/Presenters/DonationConfirmationHtmlPresenter.php]: `Donor $donor = null`

**Pull request merged**